### DISCUSSION TITLE
How can I solve BigInt turning a number into a string?

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#6007] [Q&A]
I am not sure if this is a bug or an intentional feature so this seems more appropriate. 

I am currently using MySQL for my database, and when I use a PK, BigInt, unsigned, not null field for an ID the IDs are returned as "1n", "2n" etc. I assume the n represents BigInt, but it turns the ID into a string when it shouldn't be. 

My current solution is to just replace the n with a blank and convert the string back to a number, but it seems "hacky" and not a good way of doing it. 

Why does this happen and how can I stop it from turning it into a string?  

(I am on v2.18.0)

**Comment by ryands17** [comment]
@HurricaneDancer 👋 

Prisma uses JavaScript's [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) as the output type for database BigInt. As for returning strings, Prisma doesn't output those as string but as the native `BigInt` type.

![image](https://user-images.githubusercontent.com/19697099/110305095-58ae6680-8022-11eb-8169-d0394b71041a.png)

This might be the issue when you're sending data to the frontend. I would suggest adding this to your project so that the id's are sent correctly:

```ts
BigInt.prototype.toJSON = function () {
  return this.toString()
}
```

Then they should be passed to the frontend correctly and you will get a string without the `n`.
